### ISSUE TITLE
Tons of fluentd logs brings down the ES cluster by disk exhaustion 

### ISSUE THREAD

**Issue by falfaro** [bitnami/kube-prod-runtime#193]
On a completely idle GKE cluster, and after a week, some ElasticSearch pods have entered into a crash loop because the underlying PV is full.

The ElasticSearch manifest uses 100GB by default for each PV. At the moment only one of the three ElasticSearch pods is crashing, but the other two pods will very soon start crashing as they PVs are already at 100% according to "df -k":

```
$ kubectl --namespace=kubeprod exec -it elasticsearch-logging-1 bash
Defaulting container name to elasticsearch-logging.
Use 'kubectl describe pod/elasticsearch-logging-1 -n kubeprod' to see all of the containers in this pod.
I have no name!@elasticsearch-logging-1:/$ df -k
Filesystem     1K-blocks     Used Available Use% Mounted on
overlay         98868448  6814184  92037880   7% /
tmpfs            3829072        0   3829072   0% /dev
tmpfs            3829072        0   3829072   0% /sys/fs/cgroup
/dev/sda1       98868448  6814184  92037880   7% /etc/hosts
shm                65536        0     65536   0% /dev/shm
/dev/sdb       102687672 97428408         0 100% /bitnami/elasticsearch/data
tmpfs            3829072       12   3829060   1% /run/secrets/kubernetes.io/serviceaccount
tmpfs            3829072        0   3829072   0% /sys/firmware
```

**Comment by arapulido** [comment]
It seems that this needs more investigation. The PV shouldn't be filled up that quickly

**Comment by jbianquetti-nami** [comment]
Fluentd data fills data volumes of ES. 

That's caused by  fluent container settings which set rules to monitor all containers STDIN/STOUT/log files, including himself, and therefore causing a write-read-write-read loop on log files.

related PR: https://github.com/bitnami/kube-prod-runtime/pull/205

**Comment by falfaro** [comment]
It is not enough with changing `fluentd` to only log `ERROR` and `FATAL`. We will probably need to implement some pre-processing to ignore log entries from the `fluentd` containers in the first place.

**Comment by jjo** [comment]
Went down comparing our internal fluentd-es configs with BKPR ones, finding
nothing relevant ...  but then realized we're using an upstream image instead
(`gcr.io/google-containers/fluentd-elasticsearch:v2.0.1`) which nicely comes with the
following **embedded** `/etc/fluent/fluent.conf`:
```
# This is the root config file, which only includes components of the actual configuration
# Do not collect fluentd's own logs to avoid infinite loops.
<match fluent.**>
  type null
</match>

@include /etc/fluent/config.d/*.conf
```

So I guess what triggered this issue was moving off this upstream image,
should be easily solvable by adding above conf as e.g. `avoid_fluentd.input.conf`
(explicitly starting the filename with `a*` to be loaded as 1st one).

**Comment by falfaro** [comment]
Thanks @jjo for hunting the missing piece. I've created PR #220 instead of #205.

**Comment by sameersbn** [comment]
Closed by #220
